test/lib/pgBackRestTest/Module/Storage/StorageFilterCipherBlockTest.pm

```diff
@@ -0,0 +1,221 @@
+
+        $oEncryptFileIo = $self->testResult(
+            sub {new pgBackRest::Storage::Filter::CipherBlock(
+                $oDriver->openWrite($strFileBinEncrypt), $strCipherType, $tCipherKey)},
+            '[object]', '    new write encrypt');
+
+        $self->testResult(sub {$oEncryptFileIo->write(\$tContent)}, length($tContent), '    write encrypted');
+        $self->testResult(sub {$oEncryptFileIo->close()}, true, '    close');
+        $self->testResult(
+            sub {sha1_hex(${storageTest()->get($strFileBinEncrypt)}) ne $strFileBinHash}, true, '    check sha1 different');
+
+        my $oEncryptBinFileIo = $self->testResult(
+            sub {new pgBackRest::Storage::Filter::CipherBlock(
+                $oDriver->openRead($strFileBinEncrypt), $strCipherType, $tCipherKey,
+                {strMode => STORAGE_DECRYPT})},
+            '[object]', 'new read encrypted bin file');
+
+        $self->testResult(sub {$oEncryptBinFileIo->read(\$tBuffer, 16777216)}, 16777216, '    read 16777216 bytes');
+        $self->testResult(sub {sha1_hex($tBuffer)}, $strFileBinHash, '    check sha1 same as original');
+        $self->testResult(sub {$oEncryptBinFileIo->close()}, true, '    close');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        undef($tBuffer);
+
+        $self->storageTest()->put($strFile, $strFileContent);
+
+        executeTest(
+            "openssl enc -k ${tCipherKey} -md sha1 -aes-256-cbc -in ${strFile} -out ${strFileEncrypt}");
+
+        $oEncryptFileIo = $self->testResult(
+            sub {new pgBackRest::Storage::Filter::CipherBlock(
+                $oDriver->openRead($strFileEncrypt), $strCipherType, $tCipherKey,
+                {strMode => STORAGE_DECRYPT})},
+            '[object]', 'read file encrypted by openssl');
+
+        $self->testResult(sub {$oEncryptFileIo->read(\$tBuffer, 16)}, 8, '    read 8 bytes');
+        $self->testResult(sub {$oEncryptFileIo->close()}, true, '    close');
+        $self->testResult(sub {$tBuffer}, $strFileContent, '    check content same as original');
+
+        $self->storageTest()->remove($strFile);
+        $self->storageTest()->remove($strFileEncrypt);
+
+        $oEncryptFileIo = $self->testResult(
+            sub {new pgBackRest::Storage::Filter::CipherBlock(
+                $oDriver->openWrite($strFileEncrypt), $strCipherType, $tCipherKey)},
+            '[object]', 'write file to be read by openssl');
+
+        $self->testResult(sub {$oEncryptFileIo->write(\$tBuffer)}, 8, '    write 8 bytes');
+        $self->testResult(sub {$oEncryptFileIo->close()}, true, '    close');
+
+        executeTest(
+            "openssl enc -d -k ${tCipherKey} -md sha1 -aes-256-cbc -in ${strFileEncrypt} -out ${strFile}");
+
+        $self->testResult(sub {${$self->storageTest()->get($strFile)}}, $strFileContent, '    check content same as original');
+    }
```

Comment:
Actually, no encrypted file should ever be 0 bytes.  So this was a problem in code.  0 byte files are now properly encrypted and decrypted and I have added tests.  Also added a test to error when decrypting an empty file.

Everything worked as it was (and your test worked), but this would have broken compatibility with openssl for 0 byte files.  See 0fe8aac05b58e6ccd37.